src/test/api_compatibility.ts

```diff
@@ -44,21 +45,21 @@ export function apiCompatibilityChecks(createTracer = () => new Tracer(), option
                     expect(() => { tracer.inject(spanContext, FORMAT_BINARY, {}); }).to.not.throw(Error);
                 });
 
-                it('should handle Spans and SpanContexts', () => {
+                options.checkInjectBehaviour ? it : it.skip('should handle Spans and SpanContexts',  () => {
                     const textCarrier = {};
                     expect(() => { tracer.inject(span, FORMAT_TEXT_MAP, textCarrier); }).to.not.throw(Error);
-                    expect(() => { tracer.inject(span.context(), FORMAT_TEXT_MAP, textCarrier); }).to.not.throw(Error);
+                    expect(() => { tracer.inject(span.context(), FORMAT_TEXT_MAP, textCarrier); });
```

Comment:
if you're removing `.to.not.throw(Error);`, then what is the expectation that's being set?

Reply:
That was not by intention.